browser/src/Editor/BufferManager.ts

```diff
@@ -148,13 +151,24 @@ export class Buffer implements IBuffer {
             Log.warn("getLines called with over 2500 lines, this may cause instability.")
         }
 
-        const lines = await this._neovimInstance.request<any>("nvim_buf_get_lines", [
+        const lines = await this._neovimInstance.request<MaybeStringArray>("nvim_buf_get_lines", [
             parseInt(this._id, 10),
             start,
             end,
             false,
         ])
-        return lines
+
+        // Neovim does not error if it is unable to get lines instead it returns an array
+        // of type [1, "an error message"], we only check the first on the assumption that
+        // that is where the number is placed by neovim
+        const isStringArray = (array: MaybeStringArray): array is string[] =>
```

Comment:
I wonder if, as you suggested, it makes sense to move this to a shared layer?

We could move this logic to `Session.ts` where we handle getting the `response` from Neovim:
https://github.com/onivim/oni/blob/0307da1691a1c1cede5219c63bf0f10a93231783/browser/src/neovim/Session.ts#L56

And check if we have an array that is:
- 2 items
- First item is a number
- Second item is a string
And in that case, we dispatch an error.

Otherwise, we'll have all these handlers having to implement their own logic for handling errors